src/components/CalculatorLine.js

```diff
@@ -1,19 +1,68 @@
-import React, {useState, useLayoutEffect} from 'react'
+import React, {useEffect, useState} from 'react'
 
 import {Row, Col, InputGroup, FormControl} from 'react-bootstrap'
 
-function createHandleChange(setFunction) {
-    return e => {
-        setFunction(e.target.value)
+
+function ReadOnlyField({placeholder, label, value, readOnly}) {
+    return (
+        <FormControl
+            placeholder={placeholder}
+            aria-label={label}
+            value={value}
+            readOnly={readOnly}
+        />
+    )
+}
+
+function InputField({placeholder, label, setFunction, type}) {
+    type = type ? type : 'number'
+    return (
+        <FormControl
+            placeholder={placeholder}
+            aria-label={label}
+            min='0'
+            type={type}
```

Comment:
```suggestion
            type={type || 'number'}
```
[Short circuit](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND) or [Nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) the rescue.
Nullish coalescing operator was recently added to Javascript